accumulator_updater/programs/accumulator_updater/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+mod macros;
+
+use anchor_lang::{
+    prelude::*,
+    solana_program::sysvar::{
+        self,
+        instructions::get_instruction_relative,
+    },
+    system_program::{
+        self,
+        CreateAccount,
+    },
+};
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+
+#[program]
+pub mod accumulator_updater {
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let whitelist = &mut ctx.accounts.whitelist;
+        whitelist.bump = *ctx.bumps.get("whitelist").unwrap();
+        Ok(())
+    }
+
+    //TODO: add authorization mechanism for this
+    pub fn add_allowed_program(
+        ctx: Context<AddAllowedProgram>,
+        allowed_program: Pubkey,
+    ) -> Result<()> {
+        let whitelist = &mut ctx.accounts.whitelist;
+        require_keys_neq!(allowed_program, Pubkey::default());
+        require!(
+            !whitelist.allowed_programs.contains(&allowed_program),
+            AccumulatorUpdaterError::DuplicateAllowedProgram
+        );
+        whitelist.allowed_programs.push(allowed_program);
+        Ok(())
+    }
+
+    /// Add new account(s) to be included in the accumulator
+    ///
+    /// * `base_account` - Pubkey of the original account the AccumulatorInput(s) are derived from
+    /// * `data` - Vec of AccumulatorInput account data
+    /// * `account_type` - Marker to indicate base_account account_type
+    /// * `account_schemas` - Vec of markers to indicate schemas for AccumulatorInputs. In same respective
+    ///    order as data
+    pub fn create_inputs<'info>(
+        ctx: Context<'_, '_, '_, 'info, CreateInputs<'info>>,
+        base_account: Pubkey,
+        data: Vec<Vec<u8>>,
+        account_type: u32,
+        account_schemas: Vec<u8>,
+    ) -> Result<()> {
```

Comment:
I think here you might want to check that the extra accounts are owned by the caller program

Reply:
the extra accounts here will be owned by the `SystemProgram` at the time they're passed in because they're uninitialized accounts and they're going to be owned by the `AccumulatorUpdater` program after they're created as PDAs